src/context/AlertContext.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState
+} from 'react'
+
+import Alert from '@/components/UI/alert'
+
+export interface AlertContentProps {
+  title?: string
+  text: string
+  variant: AlertVariant
+  position?: 'top' | 'bottom'
+  confirmFunction?: () => void
+  cancelFunction?: () => void
+  showFor?: number
+}
+
+export enum AlertVariant {
+  info,
+  security,
+  critical,
+  comment
+}
+
+interface AlertContextProps {
+  setAlert: (content: AlertContentProps) => void
+  clearAlert: () => void
+  visible: boolean
+}
+
+const alertContext = createContext<AlertContextProps>({
```

Comment:
The name of this context should be capitalised.

Reply:
Isn't camelCase fine?

Reply:
I've used camel as it's a locally scoped constant although happy to change, I'll wait until for a reply here to resolve this

Reply:
all good then, just thought that the camelcase is a little bit weird in the JSX

Reply:
Missed that I've swapped it.